### PR TITLE
Use more functional style to improve code readability

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

**Comment by wendigo** [comment]
There was a mix of functional and non-functional usages on that files:

```
if (optional.isPresent() {
   doSth(optional.get());
}
```

not it's consistent usage of `optional.ifPresent(value ->  doSth(value))`

**Review by hashhar — APPROVED** [review]
Looks good to me.

However is SqlFormatter performance sensitive? cc: @sopel39

**Comment by sopel39** [comment]
> However is SqlFormatter performance sensitive? cc: @sopel39

I don't think it is

File: core/trino-parser/src/main/java/io/trino/sql/SqlFormatter.java [hunk 1]
```diff
@@ -289,29 +287,24 @@ protected Void visitQuery(Query node, Integer indent)
...
-            }
-
-            if (node.getOffset().isPresent()) {
-                process(node.getOffset().get(), indent);
-            }
-
-            if (node.getLimit().isPresent()) {
-                process(node.getLimit().get(), indent);
-            }
+            node.getOrderBy().ifPresent(orderBy ->
```
Comment (electrum) [inline core/trino-parser/src/main/java/io/trino/sql/SqlFormatter.java:302]: Same, these are easier to read as
```java
node.getOrderBy().ifPresent(orderBy -> process(orderBy, indent));
node.getOffset().ifPresent(offset -> process(offset, indent));
node.getLimit().ifPresent(limit -> process(limit, indent));
```
No need to wrap very short lines

Comment (wendigo) [reply core/trino-parser/src/main/java/io/trino/sql/SqlFormatter.java:302]: Sure, I'll fix that :) Thanks!

File: core/trino-parser/src/main/java/io/trino/sql/SqlFormatter.java [hunk 2]
```diff
@@ -320,28 +309,23 @@ protected Void visitQuerySpecification(QuerySpecification node, Integer indent)
...
+            });
 
             builder.append('\n');
 
-            if (node.getWhere().isPresent()) {
-                append(indent, "WHERE " + formatExpression(node.getWhere().get()))
-                        .append('\n');
-            }
+            node.getWhere().ifPresent(where ->
+                    append(indent, "WHERE " + formatExpression(where)).append('\n'));
```
Comment (hashhar) [inline core/trino-parser/src/main/java/io/trino/sql/SqlFormatter.java:322]: nit
```suggestion
            node.getWhere().ifPresent(where -> append(indent, "WHERE " + formatExpression(where))
                    .append('\n'));
```

for consistency with other two-append calls below.

File: core/trino-parser/src/main/java/io/trino/sql/SqlFormatter.java [hunk 3]
```diff
@@ -350,17 +334,12 @@ protected Void visitQuerySpecification(QuerySpecification node, Integer indent)
...
-
-            if (node.getLimit().isPresent()) {
-                process(node.getLimit().get(), indent);
-            }
+            node.getOrderBy().ifPresent(orderBy ->
+                    process(orderBy, indent));
+            node.getOffset().ifPresent(offset ->
+                    process(offset, indent));
+            node.getLimit().ifPresent(limit ->
+                    process(limit, indent));
```
Comment (hashhar) [inline core/trino-parser/src/main/java/io/trino/sql/SqlFormatter.java:342]:
```suggestion
            node.getOrderBy().ifPresent(orderBy -> process(orderBy, indent));
            node.getOffset().ifPresent(offset -> process(offset, indent));
            node.getLimit().ifPresent(limit -> process(limit, indent));
```

File: core/trino-parser/src/main/java/io/trino/sql/SqlFormatter.java [hunk 4]
```diff
@@ -463,9 +441,8 @@ protected Void visitAllColumns(AllColumns node, Integer indent)
         protected Void visitTable(Table node, Integer indent)
         {
             builder.append(formatName(node.getName()));
-            if (node.getQueryPeriod().isPresent()) {
-                builder.append(" " + node.getQueryPeriod().get().toString());
-            }
+            node.getQueryPeriod().ifPresent(queryPeriod -> builder
+                    .append(" " + queryPeriod));
```
Comment (hashhar) [inline core/trino-parser/src/main/java/io/trino/sql/SqlFormatter.java:445]:
```suggestion
            node.getQueryPeriod().ifPresent(queryPeriod -> builder.append(" " + queryPeriod));
```

File: core/trino-parser/src/main/java/io/trino/sql/SqlFormatter.java [hunk 5]
```diff
@@ -549,46 +526,49 @@ protected Void visitPatternRecognitionRelation(PatternRecognitionRelation node,
                                 .collect(joining(", ")))
                         .append("\n");
             }
-            if (node.getOrderBy().isPresent()) {
-                process(node.getOrderBy().get(), indent + 1);
-            }
+            node.getOrderBy().ifPresent(orderBy ->
+                    process(orderBy, indent + 1));
```
Comment (hashhar) [inline core/trino-parser/src/main/java/io/trino/sql/SqlFormatter.java:530]:
```suggestion
            node.getOrderBy().ifPresent(orderBy -> process(orderBy, indent + 1));
```

**Review by hashhar — APPROVED** [review]
Some more. LGTM otherwise.

**Review by hashhar — APPROVED** [review]
LGTM.

**Comment by wendigo** [comment]
Thx!

**Pull request merged**